chef-utils/lib/chef-utils/platform.rb

```diff
@@ -0,0 +1,345 @@
+    # Determine if the current node is omnios
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def omnios?(node = __getnode)
+      node["platform"] == "omnios"
+    end
+
+    # Determine if the current node is openindiana
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def openindiana?(node = __getnode)
+      node["platform"] == "openindiana"
+    end
+
+    # Determine if the current node is opensolaris
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def opensolaris?(node = __getnode)
+      node["platform"] == "opensolaris"
+    end
+
+    # Determine if the current node is nexentacore
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def nexentacore?(node = __getnode)
+      node["platform"] == "nexentacore"
+    end
+
+    # Determine if the current node is aix
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def aix_platform?(node = __getnode)
+      node["platform"] == "aix"
+    end
+
+    # Determine if the current node is freebsd
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def freebsd_platform?(node = __getnode)
+      node["platform"] == "freebsd"
+    end
+
+    # Determine if the current node is openbsd
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def openbsd_platform?(node = __getnode)
+      node["platform"] == "openbsd"
+    end
+
+    # Determine if the current node is netbsd
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def netbsd_platform?(node = __getnode)
+      node["platform"] == "netbsd"
+    end
+
+    # Determine if the current node is dragonflybsd
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def dragonfly_platform?(node = __getnode)
+      node["platform"] == "dragonfly"
+    end
+
+    # Determine if the current node is MacOS.
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def mac_os_x_platform?(node = __getnode)
+      node["platform"] == "mac_os_x"
+    end
+
+    # Determine if the current node is gentoo
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def gentoo_platform?(node = __getnode)
+      node["platform"] == "gentoo"
+    end
+
+    # Determine if the current node is slackware.
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def slackware_platform?(node = __getnode)
+      node["platform"] == "slackware"
+    end
+
+    # Determine if the current node is SuSE.
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def suse_platform?(node = __getnode)
+      node["platform"] == "suse"
+    end
+
+    # Determine if the current node is OpenSuSE.
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def opensuse?(node = __getnode)
```

Comment:
this platform is EOL and we shouldn't move this into core. We do need opensuseleap_platform?